contrib/mesos/docs/discovery.md

```diff
@@ -0,0 +1,88 @@
+# Discovery
+
+## DNS
+
+### kube-dns
+
+[**kube-dns**](https://github.com/kubernetes/kubernetes/blob/release-1.1/docs/admin/dns.md) is a Kubernetes add-on that works out of the box with Kubernetes-Mesos.
+For details on usage see the implementation in the `cluster/mesos/docker` source tree.
+kube-dns provides records both for services and pods.
+
+### mesos-dns
+
+**NOTE:** There is still no support for publishing Kubernetes *services* in mesos-dns.
+
+**mesos-dns** communicates with the leading Mesos master to build a DNS record set that reflects the tasks running in a Mesos cluster as documented here: http://mesosphere.github.io/mesos-dns/docs/naming.html.
+As of Kubernetes-Mesos [release v0.7.2](https://github.com/mesosphere/kubernetes/releases/tag/v0.7.2-v1.1.5) there is experimental support in the scheduler to populate a task's *discovery-info* field in order to generate alternative/more friendly record names in mesos-dns, for *pods* only.
+
+To enable this feature, set `--mesos-generate-task-discovery=true` when launching the scheduler.
+
+The following discovery-info fields may be set using non-namespaced labels or else `k8s.mesosphere.io/discovery-XXX` attributes:
+
+* `visibility`: may be `framework`, `external`, or `cluster` (defaults to `cluster`)
+* `environment`
+* `location`
+* `name` (this alters record set generation in *mesos-dns*)
+* `version`
+
+In the case where both a label as well as an annotation are supplied the value of the attribute is observed.
+The interpretation of value of the `name` label (and `discovery-name` attribute) is a special case: the generated Mesos `discovery-info.name` value will be `${name}.${pod-namespace}.pod`; all other discovery-info values are passed through without modification.
+
+#### Example 1: Use a `name` label on a pod
+```yaml
+apiVersion: v1
+kind: ReplicationController
```

Comment:
it's generally a more useful feature for RC templates since those have auto-generated names that yield unpredictable task name's --> unpredictable mesos-dns record names. i don't think it's asking too much of our users to understand that they can apply the same technique to pods. anyone who's doing something other than toying around in kubernetes ins't using raw pods to deploy stuff - they're using RCs.